kore/src/Kore/Rewrite/Search.hs

```diff
@@ -153,9 +153,8 @@ matchWith sideCondition e1 e2 = do
                     ]
                     [Conditional.substitution predSubst]
             let simplified = merged
```

Comment:
Why is this necessary? See `mergePredicatesAndSubstitutions` (Line 162) third argument, we add `substitution merged` and `substitution simplified`, but they're the same thing.